app/models/index_section.rb

```diff
@@ -2,15 +2,23 @@ class IndexSection < ApplicationRecord
   include Orderable
 
   belongs_to :index, inverse_of: :index_sections, optional: false
-  has_many :index_posts, inverse_of: :index_section, dependent: :destroy
-  has_many :posts, through: :index_posts, dependent: :destroy
+  has_many :index_posts, inverse_of: :index_section, dependent: false # This is handled in callbacks
```

Comment:
Is there a reason not to use `dependent: :nullify` here? (Wanting to audit when this changes?)

Reply:
There is indeed a reason, and it is that we need the callbacks to handle reordering them